src/ManTradesTable.js

```diff
@@ -0,0 +1,178 @@
+        // .catch((err) => {
+        //   if (err.response) {
+        //     // check data
+        //     const err_msg = err.response.data.err;
+        //     if (!!err_msg?.form) {
+        //       setErrorTrades(
+        //         err_msg.form.reduce((res, filter) => {
+        //           const k = Object.keys(filter)[0];
+        //           const v = filter[k];
+        //           res[k] = v;
+        //           return res;
+        //         }, {})
+        //       );
+        //     } else {
+        //       setGeneralError(JSON.stringify(err_msg));
+        //     }
+        //   }
+        //   // something went wrong here
+        // });
+
+      }
+
+     handleHoldingEdit = (ticker, weight, index) => {
+
+        //   new_trades = this.state.trades
+        console.log('handle holding edit with index ', index, ticker)
+        
+        const new_trades = this.state.trades
+        if (new_trades[index].placeholder) {
+            new_trades.push({placeholder: true})
+        }
+        new_trades[index] = {placeholder: false, weight: weight, ticker:ticker}
+
+
+ 
+        
+        const total_weight = new_trades.reduce((sum, t) => {
+            if ( !isNaN(t.weight) && t.weight != '') {
+                console.log(t)
+                return parseFloat(t.weight) + sum
+            } else{
+                return sum
+            }
+        }, 0)
+
+        console.log('total weight', total_weight)
+
+
+        this.setState({trades: new_trades, 
+                        total_weight, total_weight})
+
+
+
+
+      }
+
+    render(){
+
+        const pallete = ['#0074D9', '#FF4136', '#2ECC40', '#FF851B', '#7FDBFF', '#B10DC9', '#FFDC00', '#001f3f', '#39CCCC', '#01FF70', '#85144b', '#F012BE', '#3D9970', '#111111', '#AAAAAA']
+        var data = this.state.trades.filter((t) => (!t.placeholder && !isNaN(t.weight))).map((t, index) => {
+ 
+                const slice =  {title: t.ticker, value : t.weight, color : pallete[index % pallete.length]}
+                console.log(slice)
+                return slice
+            
+        })
+        console.log('data length: ', data.length)
+
+        return (
+        <div>
+            <Typography variant="h4" >
+                Manual Enter Your Holdings to Create Your Portfolio
+            </Typography>
+            <Typography style={{color: "gray"}}>
+                Ensure your percentages add up to 100!
+            </Typography>
+        <div style={{ display: "flex", alignItems: "flex-start", marginTop:20}}>
+            <div style={{ flexGrow: "1",
+                        marginLeft: 20}}>
+            <Table>
+            <TableHead>
+                <TableRow>
+                    <TableCell>Ticker</TableCell>
+                    <TableCell>Percent</TableCell>
+                </TableRow>
+            </TableHead>
+            <TableBody>
+                {
+                    this.state.trades.map((t, index) => {
+                        if (t.placeholder){
```

Comment:
could DRY (since weight is the only prop) and just pass null for weight if placeholder is true (or refactor)